src/util/bip32.rs

```diff
@@ -181,6 +196,135 @@ impl serde::Serialize for ChildNumber {
     }
 }
 
+/// A BIP-32 derivation path.
+#[derive(Clone, PartialEq, Eq)]
+pub struct DerivationPath(Vec<ChildNumber>);
+impl_index_newtype!(DerivationPath, ChildNumber);
+
+impl From<Vec<ChildNumber>> for DerivationPath {
+    fn from(numbers: Vec<ChildNumber>) -> Self {
+        DerivationPath(numbers)
+    }
+}
+
+impl Into<Vec<ChildNumber>> for DerivationPath {
+    fn into(self) -> Vec<ChildNumber> {
+        self.0
+    }
+}
+
+impl<'a> From<&'a [ChildNumber]> for DerivationPath {
+    fn from(numbers: &'a [ChildNumber]) -> Self {
+        DerivationPath(numbers.to_vec())
+    }
+}
+
+impl ::std::iter::FromIterator<ChildNumber> for DerivationPath {
+    fn from_iter<T>(iter: T) -> Self where T: IntoIterator<Item = ChildNumber> {
+        DerivationPath(Vec::from_iter(iter))
+    }
+}
+
+impl<'a> ::std::iter::IntoIterator for &'a DerivationPath {
+    type Item = &'a ChildNumber;
+    type IntoIter = ::std::slice::Iter<'a, ChildNumber>;
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.iter()
+    }
+}
+
+impl AsRef<[ChildNumber]> for DerivationPath {
+    fn as_ref(&self) -> &[ChildNumber] {
+        &self.0
+    }
+}
+
+impl FromStr for DerivationPath {
+    type Err = Error;
+
+    fn from_str(path: &str) -> Result<DerivationPath, Error> {
+        let mut parts = path.split("/");
+        // First parts must be `m`.
+        if parts.next().unwrap() != "m" {
```

Comment:
Maybe make this an expect describing why this will never be `None`? I thought this could be a panic at first till I looked at the `split` docs.

Reply:
Hmm, I've seen `.unwrap()`s here when it's known to not panic.